core/templates/dev/head/domain/skill/SkillRightsObjectFactory.ts

```diff
@@ -17,67 +17,83 @@
  * skill rights domain objects.
  */
 
-var oppia = require('AppInit.ts').module;
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+export class SkillRights {
+  _skillId: number | string;
```

Comment:
No -- it's going to be one or the other. Figure out which one it is and use that. Don't have ambiguity.

In general we never have args in this codebase that can be one of multiple types.

Reply:
It should be a string, because all the ids are strings ans the backend is also a string. I did this because the [corresponding spec](https://github.com/oppia/oppia/blob/f8d99190f294e8c3d93fc855c5b89b88c583b264/core/templates/dev/head/domain/skill/SkillRightsObjectFactorySpec.ts#L32) has it as a number as well as a string. Should I remove the number part of the spec?

Reply:
I suspect the spec is wrong. Please take this opportunity to fix it (this is a positive consequence of introducing types!), and also notify the person who wrote that line in the spec (you can find it using the "Blame" feature).

Thanks!

Reply:
Btw, just to check, would lint checks fail if you made this typedef "string" and didn't change the spec?

Reply:
Done. 
/cc @tjiang11 



> Btw, just to check, would lint checks fail if you made this typedef "string" and didn't change the spec?

No, they won't. Even the FE tests don't fail surprisingly. Maybe they are converted to JS first so types won't affect? Not sure. But I have updated the spec nonetheless. PTAL!

Reply:
I'll start a thread with Ankita about this. Thanks for checking!

Reply:
Hi @seanlip @YashJipkate, I am also not sure why this is happening. I tried making a few changes like changing the type to number and then passing a string or directly using a number value in `this._skillId` but strangely none of those result in a failure. I found this [link](https://stackoverflow.com/questions/52082928/defining-type-of-variable-as-number-does-not-throw-error-even-if-value-is-string) which says type checking is a compile time process but I think that the call to constructor should involve check for the types in the compile time. I remember a few failures earlier where if a call to function did not involve the same number of args as specified in a function def an error was raised but I am not sure if arg types are also matched. I am looking into this further and will keep you all updated.

Reply:
Hi @seanlip @YashJipkate, so I figured out that typescript is basically not checking any of the files as it did earlier. It only the checks the files specified in `files` and `include` option in tsconfig. 

I reverted back to [older version of tsconfig](https://github.com/oppia/oppia/blob/0c9c01a8b0c218b4dfd6530757a6798f6788d1e6/tsconfig.json) before webpack was introduced and after compiling, I got a long list of around 500 errors which are introduced during conversion to Angular8 but are not being detected as typescript compilation is not being done directly but instead webpack is interfering in some way. 

@vojtechjelinek I am not sure how webpack has changed the compilation process, can you please guide us on this so that we can try to find a fix. Thanks!

[Here](https://docs.google.com/document/d/1vRhjV9eW1ZpUHh0oHlpRC66YfOl9kvTerbr6uFxVl1s/edit?usp=sharing) is the error log.